C7/Game.cs

```diff
@@ -436,8 +448,15 @@ private void UnitButtonPressed(string buttonName)
 		}
 		else if (buttonName.Equals("buildCity"))
 		{
-			PopupOverlay popupOverlay = GetNode<PopupOverlay>("CanvasLayer/PopupOverlay");
-			popupOverlay.ShowPopup("buildCity", PopupOverlay.PopupCategory.Advisor);
+			EngineStorage.animTracker.startAnimation(
+				OS.GetTicksMsec(),
+				CurrentlySelectedUnit.guid,
+				MapUnit.AnimatedAction.BUILD,
+				(unitGUID, action) => {
+					PopupOverlay popupOverlay = GetNode<PopupOverlay>("CanvasLayer/PopupOverlay");
+					popupOverlay.ShowPopup("buildCity", PopupOverlay.PopupCategory.Advisor);
+					return false;
+				});
```

Comment:
Unfortunately I programmed all that while misremembering how the original game works. In fact the only reason I gave AnimationTracker the ability to keep completed animations around is so that the settler can be kept on the last frame of the build animation while the name input popup is open. I'm aware of the bug where the settler gets stuck on that frame if the player cancels out of the popup. The fix would be to call AnimationTracker.endAnimation when the popup is cancelled. I looked into doing that but it would require adding another signal from the popup to the main Game object and I don't know how to do that without looking it up so I didn't bother.